spec/support/connection_monitoring_and_pooling.rb

```diff
@@ -0,0 +1,370 @@
+RSpec::Matchers.define :match_events do |actual|
```

Comment:
These matchers do not point out what the difference between actual and expected results are, only that the two are different. I implemented these matchers as verifiers in https://github.com/mongodb/mongo-ruby-driver/pull/1203, you can implement a similar verifier for cmap or some other solution that highlights the differences but please no more matchers like this in the driver.

Reply:
I'm not actually convinced that this is any better than what we already have; `verify_operation_result` just prints out the difference without context, e.g. `expected 2; got 3` when a single field is different in a list of documents from a cursor, and I find that harder to debug than what I implemented here. That said, you seem to feel much more strongly than I do on this particular issue, so I'll implement the changes as requested.